src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -632,6 +632,94 @@ private void CleanUp()
         #endregion IDisposable Members
     }
 
+    /// <summary>
+    /// Implements ConvertTo-CliXml command.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2243:AttributeStringLiteralsShouldParseCorrectly")]
+    [Cmdlet(VerbsData.ConvertTo, "CliXml", HelpUri = "")]
+    public sealed class ConvertToClixmlCommand : PSCmdlet
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets input objects to be converted to CliXml object.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets depth of serialization.
+        /// </summary>
+        [Parameter]
+        [ValidateRange(1, int.MaxValue)]
+        public int Depth { get; set; } = 1;
+
+        #endregion Parameters
+
+        #region Private Members
+
+        private readonly List<PSObject> _inputObjectBuffer = new();
+
+        #endregion Private Members
+
+        #region Overrides
+
+        /// <summary>
+        /// Process record.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            _inputObjectBuffer.Add(InputObject);
+        }
+
+        /// <summary>
+        /// End Processing.
+        /// </summary>
+        protected override void EndProcessing()
+        {
+            if (_inputObjectBuffer.Count > 0)
+            {
+                // If single input object don't wrap in array
+                object source = _inputObjectBuffer.Count == 1
+                    ? _inputObjectBuffer[0]
+                    : _inputObjectBuffer.ToArray();
```

Comment:
I guess this correlate with my previous comment.

Reply:
This is pretty much identical to how `ConverTo-Json` works, which forces singleton item array to be just one item returned.

Reply:
Just an aside: it's understandable to think the `ConvertFrom-Json` parameter name is `-AsArray` (I'm tempted to think that every time), but it's actually `-NoEnumerate`, and it doesn't _ensure_ that the output is an array; it only outputs an array (as a whole) if the conversion result happens to be an array. By contrast, `ConvertTo-Json`'s `-AsArray` currently _unconditionally constructs an array wrapper_, which I think is problematic with direct use of `-InputObject` - see #10952

Reply:
Then this code doesn't make sense here.

Reply:
Thanks @mklement0 for explanation and reminding me, I often get those parameters mixed up in my head 😄.

In this case not sure what behaviour we should opt in for this cmdlet. @iSazonov what do you suggest instead so this makes sense?

Reply:
We have no need to force output in an array. 
We should get the same result as Export-Clixml output.

Reply:
The code looks OK to me for the direct `-InputObject` argument case, but `Export-CliXml` actually works differently with _pipeline_  input (I haven't looked a how and why):

E.g., the following doesn't serialize as an array, but as two top-level integers:

```powershell
 1, 2 | Export-Clixml  temp.xml; Get-Content temp.xml
```

```
<Objs Version="...">
  <I32>1</I32>
  <I32>2</I32>
</Objs>
```

Reply:
I think this is quite tricky because `PSSerializer.Serialize` calls `WriteOneTopLevelObject`, so in this case all the input is collected into `EndProcessing` and it gets forced to an array. Will need to see how `Export-CliXml` does it, which btw is not using `PSSerializer` APIs 😄.

Reply:
The internal APIs that `Export-CliXml` use basically serialize each input object one at a time then writes the root element with `serialize.Done()` at the end. The `PSSerializer.Serialize` API however writes one whole top level object, so basically its not possible it seems to get the same behaviour as is.

Reply:
What I think will be needed is a new overload for `PSSerializer.Serialize` that serializes each input one at a time instead of everything at once like with existing code. Will try that later.

Reply:
Perhaps all these cmdlets should be based on the same api. to work identically. Then there are two ways. Just use the internal api now and switch to the public api later. Or make a public api now and migrate the rest of the cmdlets later.
Perhaps first way is more simple.